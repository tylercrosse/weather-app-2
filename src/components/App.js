import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { geocode } from "../ducks/locations";
import { fetchForecast } from "../ducks/weather";
import { showSearch, hideSearch } from "../ducks/ui";
import SearchBar from "./SearchBar";
import CurrentWeather from "./current/CurrentWeather";
import Forecast from "./forecast/Forecast";
import "./App.css";

export class App extends Component {
  render() {
    return (
      <div className="App">
        {this.props.weather.address
          ? <CurrentWeather
              showSearch={this.props.showSearch}
              hideSearch={this.props.hideSearch}
              ui={this.props.ui}
              locations={this.props.locations}
              weather={this.props.weather}
              geocode={this.props.geocode}
              fetchForecast={this.props.fetchForecast}
            />
          : <div className="search-no_current">
              <SearchBar geocode={this.props.geocode} />
            </div>}
        {this.props.weather.address && <Forecast />}
      </div>
    );
  }
}

App.propTypes = {
  ui: PropTypes.object.isRequired,
  locations: PropTypes.object.isRequired,
  weather: PropTypes.object.isRequired,
  showSearch: PropTypes.func.isRequired,
  hideSearch: PropTypes.func.isRequired,
  geocode: PropTypes.func.isRequired,
  fetchForecast: PropTypes.func.isRequired,
};

export const mapStateToProps = state => ({
  ui: state.ui,
  locations: state.locations,
  weather: state.weather,
});

export default connect(mapStateToProps, {
  geocode,
  fetchForecast,
  showSearch,
  hideSearch
})(App);
