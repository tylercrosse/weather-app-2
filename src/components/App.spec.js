import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { App, mapStateToProps } from "./App";

const setup = propOverrides => {
  const props = Object.assign(
    {
      ui: {},
      locations: {},
      weather: { address: 'Seattle, WA'},
      geocode: jest.fn(),
      fetchForecast: jest.fn(),
      showSearch: jest.fn(),
      hideSearch: jest.fn(),
    },
    propOverrides
  );

  const component = <App {...props} />;
  const wrapper = shallow(component);

  return {
    props,
    wrapper
  };
};

describe("<App />", () => {
  it("should render correctly with no conditonals", () => {
    const { wrapper } = setup();
    expect(toJson(wrapper)).toMatchSnapshot("no conditionals");
  });

  it("should render correctly with recent searches", () => {
    const { wrapper } = setup({
      locations: {
        ChIJVTPokywQkFQRmtVEaUZlJRA: {
          time: 1499098513,
          id: 'ChIJVTPokywQkFQRmtVEaUZlJRA',
          address: 'Seattle, WA, USA',
          lat: 47.6062095,
          lng: -122.3320708
        },
        'ChIJW-T2Wt7Gt4kRKl2I1CJFUsI': {
          time: 1499061460,
          id: 'ChIJW-T2Wt7Gt4kRKl2I1CJFUsI',
          address: 'Washington, DC, USA',
          lat: 38.9071923,
          lng: -77.03687070000001
        },
      },
    });
    expect(toJson(wrapper)).toMatchSnapshot("with recent searches");
  });

  it("should render correctly with current weather", () => {
    const { wrapper } = setup({
      weather: {
        address: "Seattle, WA",
        currently: {
          temperature: 73,
          icon: "fog"
        }
      }
    });
    expect(toJson(wrapper)).toMatchSnapshot("with current weather");
  });

  it("should render correctly with forecast", () => {
    const { wrapper } = setup({
      weather: { daily: {} }
    });
    expect(toJson(wrapper)).toMatchSnapshot("with forecast");
  });

  it('should receive the correct props from state', () => {
    const { props } = setup();
    expect(mapStateToProps({...props}))
      .toMatchSnapshot();
  });
});
