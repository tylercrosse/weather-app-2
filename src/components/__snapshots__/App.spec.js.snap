// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should receive the correct props from state 1`] = `
Object {
  "locations": Object {},
  "ui": Object {},
  "weather": Object {
    "address": "Seattle, WA",
  },
}
`;

exports[`no conditionals 1`] = `
<div
  className="App"
>
  <CurrentWeather
    fetchForecast={[Function]}
    geocode={[Function]}
    hideSearch={[Function]}
    locations={Object {}}
    showSearch={[Function]}
    ui={Object {}}
    weather={
      Object {
        "address": "Seattle, WA",
      }
    }
  />
  <Connect(Component) />
</div>
`;

exports[`with current weather 1`] = `
<div
  className="App"
>
  <CurrentWeather
    fetchForecast={[Function]}
    geocode={[Function]}
    hideSearch={[Function]}
    locations={Object {}}
    showSearch={[Function]}
    ui={Object {}}
    weather={
      Object {
        "address": "Seattle, WA",
        "currently": Object {
          "icon": "fog",
          "temperature": 73,
        },
      }
    }
  />
  <Connect(Component) />
</div>
`;

exports[`with forecast 1`] = `
<div
  className="App"
>
  <div
    className="search-no_current"
  >
    <SearchBar
      geocode={[Function]}
    />
  </div>
</div>
`;

exports[`with recent searches 1`] = `
<div
  className="App"
>
  <CurrentWeather
    fetchForecast={[Function]}
    geocode={[Function]}
    hideSearch={[Function]}
    locations={
      Object {
        "ChIJVTPokywQkFQRmtVEaUZlJRA": Object {
          "address": "Seattle, WA, USA",
          "id": "ChIJVTPokywQkFQRmtVEaUZlJRA",
          "lat": 47.6062095,
          "lng": -122.3320708,
          "time": 1499098513,
        },
        "ChIJW-T2Wt7Gt4kRKl2I1CJFUsI": Object {
          "address": "Washington, DC, USA",
          "id": "ChIJW-T2Wt7Gt4kRKl2I1CJFUsI",
          "lat": 38.9071923,
          "lng": -77.03687070000001,
          "time": 1499061460,
        },
      }
    }
    showSearch={[Function]}
    ui={Object {}}
    weather={
      Object {
        "address": "Seattle, WA",
      }
    }
  />
  <Connect(Component) />
</div>
`;
